### PR TITLE
JP-2657: Add unit test for better coverage of recent jump detection code updat…

### PR DESCRIPTION
**Pull request by dmggh**
Closes #
Resolves [JP-2657](https://jira.stsci.edu/browse/JP-2657)

**Description**
This PR adds a unit test to more thoroughly test the recent code change in JP-2617 (“Incorrect Flagging of saturated groups adjacent to pixels with detected jumps” ) - in PR https://github.com/spacetelescope/stcal/pull/90 . Former tests were rather incomplete, and this new unit test provides additional coverage.

Checklist
- [X] Tests
- [ ] Documentation
- [X] Change log
- [ ] Milestone
- [X] Label(s)

### CODE REVIEW

File: jwst/jump/tests/test_detect_jumps.py [hunk 1]
```diff
@@ -859,6 +859,166 @@ def test_adjacent_CRs(setup_inputs):
...
+    model.data[0, 5, 2, 5] = 196.0
+    model.data[0, 6, 2, 5] = 201.0
+    model.data[0, 7, 2, 5] = 207.0
+    # ... and set 2 of the 4 neighbors to SAT, and adjust their ramps
+    model.groupdq[0, 4:, 1, 5] = SATURATED
+    model.groupdq[0, 4:, 3, 5] = SATURATED
+    model.data[0, 4:, 1, 5] = SAT_SCI
+    model.data[0, 4:, 3, 5] = SAT_SCI
+
+    # CR (F): group 4, on edge
```
Comment (hbushouse) [inline jwst/jump/tests/test_detect_jumps.py:954]: I think you mean *not* on edge here?

**Review by hbushouse — COMMENTED** [review]
Looks fine. Just one really minor comment about a comment.

**Comment by hbushouse** [comment]
CI failures are OK (codecov rarely passes). But there's still one unanswered question from the review.

File: jwst/jump/tests/test_detect_jumps.py (hunk 1 above)
Comment (dmggh) [reply jwst/jump/tests/test_detect_jumps.py:954]: Hmmm... I think it is 'on' edge. The CR is at group=4, col=2, and row=5 - which is the final row as I created the array with nrows=6.

Comment (hbushouse) [reply jwst/jump/tests/test_detect_jumps.py:954]: D'oh! You're right.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**